### PR TITLE
Fix/712 handle comments with static invocations

### LINKED ISSUE #712: Formatting not stable with comment between class and static function call
prettier: 3.4.2
pretier-plugin-java: 2.6.6

The formatting is not stable with the following code structure and needs to be run twice to get to a stable state:
```bash
$ cat initial.java
package main;

class Main {

  public static void main(String[] args) {
    return MyClass
        // comment
        .doSomething();
  }
}
$ yarn -s prettier --plugin prettier-plugin-java initial.java | tee first.java
package main;

class Main {

    public static void main(String[] args) {
        return MyClass// comment
        .doSomething();
    }
}
$ yarn -s prettier --plugin prettier-plugin-java first.java | tee second.java
package main;

class Main {

    public static void main(String[] args) {
        return MyClass.doSomething(); // comment
    }
}
```

Maybe related #592

### PR DESCRIPTION
**Pull request by clementdessoude**
## What changed with this PR:

Fix unstable formattings with comments inside fqn parts. 
#713 was an attempt to fix the #712 issue but I realized afterwards that some other edge cases were remaining

## Example

See modified unit test files to see some examples

## Relative issues or prs:

Fix #712

### CODE REVIEW

File: packages/prettier-plugin-java/test/unit-test/bug-fixes/_input.java [hunk 1]
```diff
@@ -7,6 +7,10 @@ void process(
         Map.@NonNull Entry<String, SkeletonConfiguration> entry,
         @NonNull Map<String, Object> context
     ) {}
+
+    void process() {
+        coco.tata[]()();
```
Comment (jtkiesel) [inline packages/prettier-plugin-java/test/unit-test/bug-fixes/_input.java:12]: This isn't valid Java syntax, right? Does something like this still test the edge case you were hoping to account for?
```suggestion
        coco.tata[0]();
```

Comment (clementdessoude) [reply packages/prettier-plugin-java/test/unit-test/bug-fixes/_input.java]: You're right, I was too much on what was allowed by java-parser ! I removed this test case as it won't happen in real life.
